test/extensions/filters/http/common/fuzz/filter_fuzz_test.cc

```diff
@@ -0,0 +1,159 @@
+      }
+      Buffer::OwnedImpl buffer(data.data().Get(i));
+      ENVOY_LOG_MISC(debug, "Decoding data: {} ", buffer.toString());
+      if (filter->decodeData(buffer, end_stream) != Http::FilterDataStatus::Continue) {
+        return;
+      }
+    }
+
+    if (data.has_trailers()) {
+      ENVOY_LOG_MISC(debug, "Decoding trailers: {} ", data.trailers().DebugString());
+      Http::TestHeaderMapImpl trailers = Fuzz::fromHeaders(data.trailers());
+      filter->decodeTrailers(trailers);
+    }
+  }
+
+  // This creates the filter config and runs decode.
+  void fuzz(const envoy::extensions::filters::network::http_connection_manager::v3::HttpFilter&
+                proto_config,
+            const test::fuzz::HttpData& data) {
+    try {
+      // Try to create the filter. Exit early if the config is invalid or violates PGV constraints.
+      ENVOY_LOG_MISC(info, "filter name {}", proto_config.name());
+      auto& factory = Config::Utility::getAndCheckFactoryByName<
+          Server::Configuration::NamedHttpFilterConfigFactory>(proto_config.name());
+      ProtobufTypes::MessagePtr message = Config::Utility::translateToFactoryConfig(
+          proto_config, factory_context_.messageValidationVisitor(), factory);
+      cb_ = factory.createFilterFactoryFromProto(*message, "stats", factory_context_);
+      cb_(filter_callback_);
+    } catch (const EnvoyException& e) {
+      ENVOY_LOG_MISC(debug, "Controlled exception {}", e.what());
+      return;
+    }
+
+    decode(filter_.get(), data);
+    reset();
+  }
+
+  void reset() {
+    if (filter_.get() != nullptr) {
+      filter_.get()->onDestroy();
+    }
+    filter_.reset();
+  }
+
+  NiceMock<Server::Configuration::MockFactoryContext> factory_context_;
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks_;
+  NiceMock<Http::MockFilterChainFactoryCallbacks> filter_callback_;
+  std::shared_ptr<Http::StreamDecoderFilter> filter_;
+  Http::FilterFactoryCb cb_;
+  NiceMock<Envoy::Network::MockConnection> connection_;
+  Network::Address::InstanceConstSharedPtr addr_;
+};
+
+DEFINE_PROTO_FUZZER(const test::extensions::filters::http::FilterFuzzTestCase& input) {
+  static PostProcessorRegistration reg = {[](test::extensions::filters::http::FilterFuzzTestCase*
+                                                 input,
+                                             unsigned int seed) {
+    // This ensures that the mutated configs all have valid filter names and type_urls.
+    static const std::vector<absl::string_view> filter_names = Registry::FactoryRegistry<
+        Server::Configuration::NamedHttpFilterConfigFactory>::registeredNames();
+    static const auto factories =
+        Registry::FactoryRegistry<Server::Configuration::NamedHttpFilterConfigFactory>::factories();
+    if (std::find(filter_names.begin(), filter_names.end(), input->config().name()) ==
+        std::end(filter_names)) {
+      absl::string_view filter_name = filter_names[seed % filter_names.size()];
```

Comment:
How does this work for replay of a corpus? If I have a corpus entry with a blank filter name, am I guaranteed that I will always get the same substitution in post processing? Is there some way the seed is derived from the corpus content?

Reply:
Ack, that's nice, maybe add a comment for the uninitiated.